#FP-16 Admin registration page

stewartlloydjones opened the pull request.

stewartlloyd-jones committed:
#FP-16 Admin Registration page

stewartlloyd-jones committed:
Merge branch 'main' of https://github.com/Reason0x6/FlightPub into Admin_registration_page

WillPaterson commented on src/main/resources/templates/Fragments/Forms.html:
```diff
@@ -305,4 +305,52 @@
+
+
+    <script>
+        function validateAdminRegister() {
```
Not sure why but this doesn't properly validate the inputs. 
@Reason0x6 any ideas?

WillPaterson replied:
I pushed a fix to the branch for this @stewartlloydjones
Was just an issue with how the firstname and lastname had capital Names and the validation JS didn't.

WillPaterson commented on src/main/resources/templates/Fragments/Forms.html:
```diff
@@ -305,4 +305,52 @@
+                alert("Passwords do not match");
+                return false;
+            }
+            else if(!(f["registrationcode"].value === "h38DxU9QneP1gB")){
```
This should probably be done server side

TobyChurches replied:
I think the 'registrationcode' should be verified on the server, Potentially in a different commit make the value generated by another admin account

TobyChurches commented on src/main/resources/templates/Fragments/Forms.html:
```diff
@@ -305,4 +305,52 @@
+            if(f["confirmpassword"].value == ""){ x = false; }
+            if(f["registrationcode"].value == ""){ x = false; }
+
+            if (x == "") {
```
Should this be testing if(x == false)?

WillPaterson replied:
The original code for user registration (Which I'm assuming this was based on. And that also validates correctly) has the same thing. I couldn't workout why this code wouldn't be working the same.

Commit:
Fixed issue with admin form validation.

firstname and lastname needed capitals for Name. Caused miss match between form and validation.

Reason0x6 committed:
Merge branch 'main' into Admin_registration_page
